python/cuml/dask/preprocessing/LabelEncoder.py

```diff
@@ -0,0 +1,190 @@
+
+        print(encoded.compute())
+
+        # We can assign this to a new column
+        ddf = ddf.assign(encoded=encoded)
+        print(ddf.compute())
+
+        # We can also encode more data
+        test_data = cudf.Series(['c', 'a'])
+        encoded = le.transform(dask_cudf.from_cudf(test_data, npartitions=2))
+        print(encoded.compute())
+
+        # After train, ordinal label can be inverse_transform() back to
+        # string labels
+        ord_label = cudf.Series([0, 0, 1, 2, 1])
+        ord_label = dask_cudf.from_cudf(ord_label, npartitions=2)
+
+        print(ord_label.compute())
+
+    Output:
+
+    .. code-block:: python
+
+        [0 1 2 0 0]
+
+        [0 1 2 0 0]
+
+           num_col cat_col  encoded
+        0       10       a        0
+        1       20       b        1
+        2       30       c        2
+        3       30       a        0
+        4       30       a        0
+
+        [2 0]
+
+        0    a
+        1    a
+        2    b
+        0    c
+        1    b
+        dtype: object
+
+    """
+    def __init__(self, client=None, verbose=False, **kwargs):
+        super(LabelEncoder, self).__init__(client=client,
+                                           verbose=verbose,
+                                           **kwargs)
+
+    def fit(self, y):
+        """
+        Fit a LabelEncoder (nvcategory) instance to a set of categories
+
+        Parameters
+        ----------
+        y : dask_cudf.Series
+            Series containing the categories to be encoded. It's elements
+            may or may not be unique
+
+        Returns
+        -------
+        self : LabelEncoder
+            A fitted instance of itself to allow method chaining
+
+        """
+        _classes = y.unique().compute()
+        el = first(y) if isinstance(y, Sequence) else y
+        self.datatype = ('cudf' if isinstance(el, (dcDataFrame, daskSeries))
+                         else 'cupy')
+        from cuml.preprocessing import LabelEncoder as LE
+        self._set_internal_model(LE(**self.kwargs).fit(y, _classes=_classes))
+        return self
+
+    def fit_transform(self, y, delayed=True):
+        """
+        Simultaneously fit and transform an input
+
+        This is functionally equivalent to (but faster than)
+        LabelEncoder().fit(y).transform(y)
+        """
+        return self.fit(y).transform(y, delayed=delayed)
+
+    def transform(self, y, delayed=True):
+        """
+        Transform an input into its categorical keys.
+
+        This is intended for use with small inputs relative to the size of the
+        dataset. For fitting and transforming an entire dataset, prefer
+        `fit_transform`.
+
+        Parameters
+        ----------
+        y : dask_cudf.Series
+            Input keys to be transformed. Its values should match the
+            categories given to `fit`
+
+        Returns
+        -------
+        encoded : cuPy backed Dask Array
```

Comment:
I think we return a Series in the single-GPU label encoder... any reason not to return a Series here?

Reply:
Same as before.
The indices were misaligned with trying to return a series. It seems to start from 0 for each client's output
For the case with two clients, this is the output:

```
0    0
1    1
2    2
0    0
1    0
```

Reply:
@Nanthini10, does calling `reset_index()` on the final dask_cudf Series fix this for you?

Reply:
@cjnolet Calling `reset_index` on dask-LabelEncoder after calling `_transform` did not fix this. I get the following output when I call `compute`

```
   index  0
0      0  0
1      1  1
2      2  2
0      0  0
1      1  0
```